lib/grn_ecmascript.lemon

```diff
@@ -417,16 +417,20 @@ primary_expression ::= STRING.
 primary_expression ::= BOOLEAN.
 primary_expression ::= NULL.
 
-array_literal ::= BRACKETL elision BRACKETR.
-array_literal ::= BRACKETL element_list elision BRACKETR.
+array_literal ::= BRACKETL elision BRACKETR. {
```

Comment:
Is `elision` needed?

Reply:
Yes. I think elision is needed.

If we execute as below code, occurs error in tests(`test/command/run-test.sh test/command/suite/select/filter/vector/not_equal.test` ans `equal.test`).

```
array_literal ::= BRACKETL element_list BRACKETR.

element_list ::= .
element_list ::= assignment_expression.
element_list ::= element_list COMMA assignment_expression.
```

* Error of tests

```
=======================================================================================================
[3] .
  not_equal                                                                            0.1688s [failed]
=======================================================================================================
--- (expected)
+++ (actual)
@@ -1,46 +1,13 @@
  table_create Memos TABLE_HASH_KEY ShortText
  [[0,0.0,0.0],true]
  column_create Memos tags COLUMN_VECTOR ShortText
  [[0,0.0,0.0],true]
  load --table Memos
  [
  {"_key": "groonga is very fast", "tags": ["groonga"]}
  ]
  [[0,0.0,0.0],1]
  select Memos --filter 'tags != ["rroonga"]'
- [
-   [
-     0,
-     0.0,
-     0.0
-   ],
-   [
-     [
-       [
-         1
-       ],
-       [
-         [
-           "_id",
-           "UInt32"
-         ],
-         [
-           "_key",
?                 "Sh                or      tT   e   xt"  
? [[[-22,0.0,0.0], can't find class f   objec  typ : 0 6 ]]
-         ],
-         [
-           "tags",
-           "ShortText"
-         ]
-       ],
-       [
-         1,
?         "g       roo    nga  is very fas           t",        
? #|e| Ar   umentEr  r: ca 't f nd cl     s for objec  type: 0x6
-         [
-           "groonga"
-         ]
-       ]
-     ]
-   ]
- ]
  
=======================================================================================================
[0] .
  equal                                                                                0.1766s [failed]
=======================================================================================================
--- (expected)
+++ (actual)
@@ -1,39 +1,13 @@
  table_create Memos TABLE_HASH_KEY ShortText
  [[0,0.0,0.0],true]
  column_create Memos tags COLUMN_VECTOR ShortText
  [[0,0.0,0.0],true]
  load --table Memos
  [
  {"_key": "groonga is very fast", "tags": ["groonga"]}
  ]
  [[0,0.0,0.0],1]
  select Memos --filter 'tags == ["groonga"]'
- [
-   [
-     0,
-     0.0,
-     0.0
-   ],
-   [
-     [
-       [
-         0
-       ],
-       [
-         [
-           "_id",
-           "UInt32"
-         ],
-         [
-           "_key",
?                 "Sh                or      tT   e   xt"  
? [[[-22,0.0,0.0], can't find class f   objec  typ : 0 6 ]]
-         ],
-         [
-           "tags",
?           "Sh   or      tT                  e          xt"
? #|e| ArgumentErr  : can'  find class for obj ct type: 0 6 
-         ]
-       ]
-     ]
-   ]
- ]
```

Therefore, I make the definition of `elision`.

Reply:
The error is caused by generating invalid expression.
It generates the following expression:

  * `get_value: tags`
  * `push: "groonga"`
  * `push: []`
  * `equal`

It should be the following:

  * `get_value: tags`
  * `push: ["groonga"]`
  * `equal`

I pushed array literal support to master. Could you rebase on master?

Reply:
Thank you.
I'll rebase on master.